File: problems/Math/1828. Queries on Number of Points Inside a Circle.py
#1828. Queries on Number of Points Inside a Circle
import math
from typing import List
class Solution:
    def countPoints(self, points: List[List[int]], queries: List[List[int]]) -> List[int]:
        res = []
        for i in queries:
            cur = 0 
            for j in points:
                if math.pow((j[0]-i[0]),2)+math.pow((j[1]-i[1]),2)-math.pow(i[2],2)<=0:
                    cur =cur+1
                    
            res.append(cur)
        return res
print(Solution.countPoints(Solution,[[1,1],[2,2],[3,3],[4,4],[5,5]],[[1,2,2],[2,2,2],[4,3,2],[4,3,3]]))

File: Golang/Array/1828. Queries on Number of Points Inside a Circle.go
package main

func countPoints(points [][]int, queries [][]int) []int {
	var res []int
	for i := 0; i < len(queries); i++ {
		curRes := 0
		for j := 0; j < len(points); j++ {
			row := (queries[i][0] - points[j][0]) * (queries[i][0] - points[j][0])
			column := (queries[i][1] - points[j][1]) * (queries[i][1] - points[j][1])
			if row+column <= queries[i][2]*queries[i][2] {
				curRes = curRes + 1
			}
		}
		res = append(res, curRes)
	}
	return res
}

func main() {
	points := [][]int{{1, 1}, {2, 2}, {3, 3}, {4, 4}, {5, 5}}
	queries := [][]int{{1, 2, 2}, {2, 2, 2}, {4, 3, 2}, {4, 3, 3}}
	output := countPoints(points, queries)
	for i := 0; i < len(output); i++ {
		println(output[i])
	}
}
